### PR TITLE
Fix Golang lint warnings and issues

### PR DESCRIPTION
**Pull request by nmagnezi**

### CODE REVIEW

File: pkg/asset/ignition/bootstrap_ignition.go [hunk 1]
```diff
@@ -52,7 +52,7 @@ var _ asset.Asset = (*BootstrapIgnition)(nil)
 
 // Name returns the human-friendly name of the asset.
 func (i *BootstrapIgnition) Name() string {
-	return "Bootstrap Ignition"
+	return "Bootstrap IgnitionInterface"
```
Comment (danielerez) [inline pkg/asset/ignition/bootstrap_ignition.go:55]: should remain 'Ignition'

File: pkg/asset/ignition/install_ignition.go [hunk 2]
```diff
@@ -42,7 +42,7 @@ var _ asset.Asset = (*InstallIgnition)(nil)
 
 // Name returns the human-friendly name of the asset.
 func (i *InstallIgnition) Name() string {
-	return "Install Ignition"
+	return "Install IgnitionInterface"
```
Comment (danielerez) [inline pkg/asset/ignition/install_ignition.go:45]: should remain 'Ignition'

File: pkg/ignition/ignition.go [hunk 3]
```diff
@@ -12,25 +12,21 @@ import (
...
 )
 
-// Added to manage ignition versions in one file
-var EmptyIgnitionConfig = types.IgnitionConfig{}
-var EmptyIgnition = types.Config{}
-
-//go:generate mockgen -source=ignition.go -package=ignition -destination=mock_ignition.go
-type Ignition interface {
+//go:generate mockgen -source=Ignition.go -package=Ignition -destination=mock_ignition.go
+type IgnitionInterface interface {
```
Comment (danielerez) [inline pkg/ignition/ignition.go:16]: the naming is a bit weird:/ do we have to export the struct? we don't export in coreos.go for example, or in [release.go](https://github.com/openshift/assisted-service/blob/c4a6395c0059890d083379e0bc28f34e072f669d/internal/oc/release.go#L58-L59)

File: pkg/asset/ignition/install_ignition.go (hunk 2 above)
Comment (nmagnezi) [reply pkg/asset/ignition/install_ignition.go]: done

**Review by danielerez — APPROVED** [review]

**Pull request merged**